### PR TITLE
Features/three story tabs

### PR DESCRIPTION
**Pull request by other91**
The first implementation

### CODE REVIEW

File: ITRStartTask/AppLibrary/ViewControllersFactory.swift [hunk 1]
```diff
@@ -9,20 +9,59 @@
 import UIKit
 
 class ViewControllersFactory {
-    static func getStoriesViewController() -> UIViewController {
+    static func getStoryViewController(storyItem: StoryItemProtocol) -> UIViewController {
+        let view: StoryViewController = StoryViewController.xibInstance()
+        let presenter = StoryPresenter(view: view, model: storyItem, commentsDataProvider: CommentsDataProvider())
```
Comment (asleptsov-itr) [inline ITRStartTask/AppLibrary/ViewControllersFactory.swift]: let provider: Protocol = ProtocolImplementation()

File: ITRStartTask/AppLibrary/ViewControllersFactory.swift [hunk 2]
```diff
@@ -9,20 +9,59 @@
...
+        return view
+    }
+
+    static func getStoryCategoriesTabBarController() -> UIViewController {
+        let categoriesView = UITabBarController()
+        categoriesView.title = NSLocalizedString("Stories", comment: "")
+
+        let topStoriesView = self.getTopStoriesViewController()
+        topStoriesView.tabBarItem = UITabBarItem(title: NSLocalizedString("TopStories", comment: ""),
+                                                 image: UIImage.init(named: "Like"),
```
Comment (asleptsov-itr) [inline ITRStartTask/AppLibrary/ViewControllersFactory.swift]: UIImage(named: "Like")

File: ITRStartTask/AppLibrary/ViewControllersFactory.swift [hunk 3]
```diff
@@ -9,20 +9,59 @@
...
+    static func getStoryViewController(storyItem: StoryItemProtocol) -> UIViewController {
+        let view: StoryViewController = StoryViewController.xibInstance()
+        let presenter = StoryPresenter(view: view, model: storyItem, commentsDataProvider: CommentsDataProvider())
+        view.presenter = presenter
+
+        return view
+    }
+
+    static func getStoryCategoriesTabBarController() -> UIViewController {
+        let categoriesView = UITabBarController()
```
Comment (asleptsov-itr) [inline ITRStartTask/AppLibrary/ViewControllersFactory.swift]: categoriesView -> categoriesTabBarController

File: ITRStartTask/AppLibrary/ViewControllersFactory.swift [hunk 4]
```diff
@@ -9,20 +9,59 @@
...
+        categoriesView.title = NSLocalizedString("Stories", comment: "")
+
+        let topStoriesView = self.getTopStoriesViewController()
```
Comment (asleptsov-itr) [inline ITRStartTask/AppLibrary/ViewControllersFactory.swift]: topStoriesView -> topStoriesViewController

File: ITRStartTask/AppLibrary/ViewControllersFactory.swift [hunk 5]
```diff
@@ -9,20 +9,59 @@
...
+
+        categoriesView.setViewControllers([topStoriesView, newStoriesView, bestStoriesView], animated: true)
+
+        return categoriesView
+    }
+
+    private static func getTopStoriesViewController() -> UIViewController {
         let view: StoriesViewController = StoriesViewController.xibInstance()
-        let presenter = StoriesPresenter(view: view, storiesDataProvider: StoriesDataProvider())
+        let presenter = StoriesPresenter(view: view, storiesDataProvider: StoriesDataProvider(category: .top))
```
Comment (asleptsov-itr) [inline ITRStartTask/AppLibrary/ViewControllersFactory.swift]: see comment above...
let dataProvider: ... = StoriesDataProvider(category: .top)

File: ITRStartTask/StoriesList/Models/StoriesDataProvider.swift [hunk 6]
```diff
@@ -9,32 +9,52 @@
...
                 DispatchQueue.main.async {
                     onFailure(error)
                 }
             })
-        }, onFailure: { (error) in
+        }
 
+        let failure: (Error) -> Void  = { (error) in
             DispatchQueue.main.async {
                 onFailure(error)
```
Comment (asleptsov-itr) [inline ITRStartTask/StoriesList/Models/StoriesDataProvider.swift:46]: ???
self?.onFailure(error)

File: ITRStartTask/AppLibrary/ViewControllersFactory.swift (hunk 1 above)
Comment (other91) [reply ITRStartTask/AppLibrary/ViewControllersFactory.swift]: fixed

File: ITRStartTask/AppLibrary/ViewControllersFactory.swift (hunk 2 above)
Comment (other91) [reply ITRStartTask/AppLibrary/ViewControllersFactory.swift]: fixed

File: ITRStartTask/AppLibrary/ViewControllersFactory.swift (hunk 3 above)
Comment (other91) [reply ITRStartTask/AppLibrary/ViewControllersFactory.swift]: fixed

File: ITRStartTask/AppLibrary/ViewControllersFactory.swift (hunk 4 above)
Comment (other91) [reply ITRStartTask/AppLibrary/ViewControllersFactory.swift]: fixed

File: ITRStartTask/AppLibrary/ViewControllersFactory.swift (hunk 5 above)
Comment (other91) [reply ITRStartTask/AppLibrary/ViewControllersFactory.swift]: fixed

File: ITRStartTask/StoriesList/Models/StoriesDataProvider.swift (hunk 6 above)
Comment (other91) [reply ITRStartTask/StoriesList/Models/StoriesDataProvider.swift:46]: parameter. Fixed

File: ITRStartTask/AppLibrary/ViewControllersFactory.swift [hunk 7]
```diff
@@ -11,55 +11,64 @@ import UIKit
...
         view.presenter = presenter
 
         return view
     }
 
     private static func getBestStoriesViewController() -> UIViewController {
         let view: StoriesViewController = StoriesViewController.xibInstance()
-        let presenter = StoriesPresenter(view: view, storiesDataProvider: StoriesDataProvider(category: .best))
+        let dataProvider: StoriesDataProviderProtocol = StoriesDataProvider(category: .best)
+
```
Comment (asleptsov-itr) [inline ITRStartTask/AppLibrary/ViewControllersFactory.swift]: remove these spaces

**Pull request merged**